components/eamxx/src/control/atmosphere_driver.cpp

```diff
@@ -1071,10 +1076,10 @@ void AtmosphereDriver::set_initial_conditions ()
           const auto& fname = fid.name();
           if (ic_pl.isParameter(fname) and ic_pl.isType<double>(fname)) {
             initialize_constant_field(fid, ic_pl);
-            fields_inited[grid_name].push_back(fname);
           } else {
             this_grid_ic_fnames.push_back(fname);
           }
+	  fields_inited[grid_name].push_back(fname);
```

Comment:
We were only using `fields_inited` for fields set as params, but I think it makes sense to use for any field we init, including from file. I use this to verify random perturbation was requested from an appropriate field.

Reply:
Super minor point - if you're using fields_inited for everything, why not call fields_inited 1x just after all these if/elseif conditions instead of repeating the same line in each if/elseif case?

Reply:
Yeah, maybe we can. There are some fields that are skipped, but that should be easy to account for. I'll look at it.

Reply:
Whatever's easiest and cleanest. Just something to consider.

Reply:
I'm going to leave this as is since Luca is reworking this whole framework anyways.